### PR TITLE
adding extensions so we have a common framework

### PR DESCRIPTION
**Pull request by mstaron17**

### CODE REVIEW

File: client/public.go [hunk 1]
```diff
@@ -110,6 +110,26 @@ type Profile struct {
 	AccessToken          string            `json:"-"`
 	Landing              string            `json:"landing" validation:"required"`
 	Program              string            `json:"program" validation:"required"`
+	Extensions           *[]*ExtensionData `json:"extensions,omitempty" sub-key:"-"`
```
Comment (mstaron17) [inline client/public.go]: copied this "sub-key" from admin, don't know if it is needed or what

Comment (flinnb) [reply client/public.go]: It's the substitution key to use when generating a map for custom emails.  Definitely don't need it here...

**Review by flinnb — APPROVED** [review]

File: client/public.go (hunk 1 above)
Comment (mstaron17) [reply client/public.go]: removed the subkey

File: handlers/static/static_test.go [hunk 2]
```diff
@@ -69,7 +69,8 @@ func TestHandleStaticALB(t *testing.T) {
 
 		require.NoError(t, err)
 		require.NotNil(t, r)
-
+		// adding skip here since github is failing this and no one can produce locally
+		t.Skip()
```
Comment (flinnb) [inline handlers/static/static_test.go]: You can add a comment to `t.Skip()`:

```go
		t.Skip("This seems to fail on github actions, we don't know why yet")
```
for example

Comment (flinnb) [reply handlers/static/static_test.go]: That way it shows up in test output also

**Pull request merged**